test/unit/query-interface/delete.test.ts

```diff
@@ -39,4 +39,88 @@ describe('QueryInterface#delete', () => {
 
     expect(firstCall.args[1]?.bind).to.be.undefined;
   });
+
+  it('returns query with `options.returning` set to true', async () => {
+    const stub = sinon.stub(sequelize, 'queryRaw');
+    const instance = new User();
+
+    await sequelize.getQueryInterface().delete(
+      instance,
+      User.tableName,
+      { id: ':id' },
+      {
+        returning: true,
+        replacements: {
+          id: '123',
+        },
+      },
+    );
+
+    expect(stub.callCount).to.eq(1);
+    const firstCall = stub.getCall(0);
+    expectsql(firstCall.args[0] as string, {
+      default: 'DELETE FROM `Users` WHERE `id` = \':id\'',
+      postgres: `DELETE FROM "Users" WHERE "id" = ':id' RETURNING "id","firstName"`,
```

Comment:
Is it better to leave it as such to run for all dialects or bring these tests under an if condition to run only for postgres?

Reply:
I think something like you have is fine. Please make use of default however (so db2, ibmi, snowflake etc is not necessary it seems).